airflow/providers/amazon/aws/hooks/emr.py

```diff
@@ -20,16 +20,25 @@
 import json
 import warnings
 from time import sleep
-from typing import Any, Callable
+from typing import TYPE_CHECKING, Any, Callable
 
 from botocore.exceptions import ClientError
 
 from airflow.compat.functools import cached_property
 from airflow.exceptions import AirflowException, AirflowNotFoundException
-from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook, AwsGenericHook
+
+if TYPE_CHECKING:
+    from mypy_boto3_emr import EMRClient
+    from mypy_boto3_emr.literals import ClusterStateType
+    from mypy_boto3_emr.type_defs import (
+        ListClustersOutputTypeDef,
+        RunJobFlowInputRequestTypeDef,
+        RunJobFlowOutputTypeDef,
+    )
 
 
-class EmrHook(AwsBaseHook):
+class EmrHook(AwsGenericHook[EMRClient]):
```

Comment:
```suggestion
class EmrHook(AwsGenericHook["EMRClient"]):
```

Unfortunately Python is not _that_ smart